homeassistant/components/lgthinq/entity_helpers.py

```diff
@@ -0,0 +1,312 @@
+
+    AIR_FAN_OPERATION_MODE = "air_fan_operation_mode"
+    AIR_PURIFIER_OPERATION_MODE = "air_purifier_operation_mode"
+    DEHUMIDIFIER_OPERATION_MODE = "dehumidifier_operation_mode"
+    HUMIDIFIER_OPERATION_MODE = "humidifier_operation_mode"
+
+
+OPERATION_SWITCH_DESC: dict[Operation, ThinQSwitchEntityDescription] = {
+    Operation.AIR_FAN_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=Operation.AIR_FAN_OPERATION_MODE,
+        icon="mdi:power",
+        name="Power",
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=Operation.AIR_FAN_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+    Operation.AIR_PURIFIER_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=Operation.AIR_PURIFIER_OPERATION_MODE,
+        icon="mdi:power",
+        name="Power",
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=Operation.AIR_PURIFIER_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+    Operation.DEHUMIDIFIER_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=Operation.DEHUMIDIFIER_OPERATION_MODE,
+        icon="mdi:power",
+        name="Power",
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=Operation.DEHUMIDIFIER_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+    Operation.HUMIDIFIER_OPERATION_MODE: ThinQSwitchEntityDescription(
+        key=Operation.HUMIDIFIER_OPERATION_MODE,
+        icon="mdi:power",
+        name="Power",
+        translation_key="operation_power",
+        property_info=PropertyInfo(
+            key=Operation.HUMIDIFIER_OPERATION_MODE,
+            value_converter=value_to_power_state_converter,
+        ),
+    ),
+}
+
+AIR_PURIFIER_FAN_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[Operation.AIR_FAN_OPERATION_MODE],
+)
+AIRPURIFIER_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[Operation.AIR_PURIFIER_OPERATION_MODE],
+)
+DEHUMIDIFIER_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[Operation.DEHUMIDIFIER_OPERATION_MODE],
+)
+HUMIDIFIER_SWITCH: tuple[ThinQSwitchEntityDescription, ...] = (
+    OPERATION_SWITCH_DESC[Operation.HUMIDIFIER_OPERATION_MODE],
+)
+
+
+# The entity escription map for each device type.
+ENTITY_MAP = {
+    DeviceType.AIR_PURIFIER_FAN: {Platform.SWITCH: AIR_PURIFIER_FAN_SWITCH},
+    DeviceType.AIR_PURIFIER: {Platform.SWITCH: AIRPURIFIER_SWITCH},
+    DeviceType.DEHUMIDIFIER: {Platform.SWITCH: DEHUMIDIFIER_SWITCH},
+    DeviceType.HUMIDIFIER: {Platform.SWITCH: HUMIDIFIER_SWITCH},
+}
+
+UNIT_CONVERSION_MAP: dict[str, str] = {
+    "F": UnitOfTemperature.FAHRENHEIT,
+    "C": UnitOfTemperature.CELSIUS,
+}
+
+
+class ThinQEntity(CoordinatorEntity, Generic[ThinQEntityDescriptionT]):
```

Comment:
Why is it generic? You can just overwrite the base entity_description type

Reply:
Removed.